src/mono/configure.ac

```diff
@@ -6795,12 +6795,49 @@ AC_COMPILE_IFELSE(
 		AC_MSG_RESULT(no)
 	])
 
+# for icu shim
+if test x$with_core = xonly; then
+	ORIG_CPPFLAGS=$CPPFLAGS
+	CPPFLAGS="$CPPFLAGS -Wno-reserved-id-macro -Wno-documentation -Wno-documentation-unknown-command -Wno-switch-enum -Wno-covered-switch-default -Wno-covered-switch-default -Wno-extra-semi-stmt -Wno-unknown-warning-option"
+
+	if test x$target_osx = xyes; then
+		CPPFLAGS="$CPPFLAGS -I/usr/local/opt/icu4c/include"
+		AC_CHECK_LIB(icucore, ucol_open, [],
+			[AC_MSG_ERROR([Cannot find libicucore, skipping build for System.Globalization.Native. .NET globalization is not expected to function.])])
+		AC_DEFINE(ICU_LIBS, -licucore, [adding -licucore])
+		AC_CHECK_HEADER(unicode/utypes.h, [have_sys_icu=yes], [have_sys_icu=no])
+		if test x$have_sys_icu = xyes; then
+			ICU_CFLAGS="-DOSX_ICU_LIBRARY_PATH=AS_ESCAPE(\"/usr/lib/libicucore.dylib\", '\"') -DTARGET_UNIX -DU_DISABLE_RENAMING -I/usr/local/opt/icu4c/include -Wno-reserved-id-macro -Wno-documentation -Wno-documentation-unknown-command -Wno-switch-enum -Wno-covered-switch-default -Wno-covered-switch-default -Wno-extra-semi-stmt -Wno-unknown-warning-option -Wno-deprecated-declarations"
```

Comment:
Again, where are these flags actually coming from? Isn't this just the new `CPPFLAGS` with a couple additions?